manager/src/test/java/org/gingersnapproject/metrics/MetricsResourceTest.java

```diff
@@ -0,0 +1,32 @@
+package org.gingersnapproject.metrics;
+
+import io.gingersnapproject.metrics.micrometer.TimerMetric;
+import io.quarkus.test.junit.QuarkusTest;
+import org.hamcrest.CoreMatchers;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+
+@QuarkusTest
+public class MetricsResourceTest {
+
+    @Test
+    public void testMetricsEndpoint() {
+        String rule = "test-rule";
+        String key = "test-key";
+        given()
+                .pathParam("rule", rule).and().pathParam("key", key)
+                .when().get("/{rule}/{key}")
+                .then().statusCode(404);
+
+        given()
+                .get("/q/metrics")
+                .then().statusCode(200)
+                .body(CoreMatchers.containsString(TimerMetric.CACHE_LOCAL_HIT.metricName()),
+                        CoreMatchers.containsString(TimerMetric.CACHE_REMOTE_HIT.metricName()),
+                        CoreMatchers.containsString(TimerMetric.CACHE_LOCAL_MISS.metricName()),
+                        CoreMatchers.containsString(TimerMetric.CACHE_REMOTE_MISS.metricName()),
+                        CoreMatchers.containsString(TimerMetric.CACHE_ERROR.metricName()));
```

Comment:
Can we add an additional test to check that these values actually increase? We could use a h2 container as the DB

Reply:
Or maybe just MySQL given that we already have the driver.

https://github.com/gingersnap-project/db-syncer/blob/main/src/test/java/io/gingersnapproject/MySQLResources.java